### DISCUSSION TITLE
Enable interfaces to support more versions of Solidity

### DISCUSSION THREAD

**Discussion by scorpion9979** [PaulRBerg/prb-contracts#39] [Ideas]
Currently the smart contract interfaces are locked to higher versions of Solidity, which makes it difficult for developers using older versions of Solidity to integrate them.

While I do understand that it's a best-practice for developers to always use the latest version of Solidity, I personally often find myself having to use an older version when integrating with 3rd party smart contract libraries that are locked to older versions.

To improve compatibility and ease of integration, I suggest modifying the Solidity version pragma in interfaces in order to make it much wider than it currently is.

**Comment by PaulRBerg** [comment]
I cannot lower the pragmas in the interfaces, unfortunately.

The interfaces contain custom errors, which have been introduced in [Solidity v0.8.4](https://blog.soliditylang.org/2021/04/21/solidity-0.8.4-release-announcement/).

I have [considered](https://twitter.com/PaulRBerg/status/1587041303015165952) the idea of moving all errors to a shared a library `Errors.sol`, but that doesn't work because [Solidity doesn't include events emitted outside of the contract in the contract ABI](https://github.com/ethereum/solidity/issues/13086#issuecomment-1387257613).
